src/audio/module_adapter/module_adapter.c

```diff
@@ -670,20 +700,26 @@ int module_adapter_copy(struct comp_dev *dev)
 	}
 
 	if (mod->simple_copy) {
-		comp_update_buffer_consume(source_c, mod->input_buffers[0].consumed);
+		i = 0;
+		list_for_item(blist, &dev->bsource_list) {
```

Comment:
`for (i = 0; i < num_input_buffers; i++)`
BTW, doesn't the compiler complain about potentially uninitialised `source_c[]`?

Reply:
I cant use the suggsted for llop lyakh, num_input_buffers is the count of active sources but I acquire all buffers irrespective of whether they are active or not. And if there are N sources, I'd assume that source_c couldnt possibly be NULL

Reply:
I mean the above `if (...) continue;` - that means, that it is possible, that not each iteration over the list will add a buffer to the array. You're right, that `num_input_buffers` is different, sorry for a confusion. But then you need a different variable to store the number of buffers that have been assigned to the array.

Reply:
@lyakh but I do add a buffer to the array before the if () continue isnt it? Are you questioning the call to consume below? This is harmless isnt it because mod->input_buffers[i].consumed will be 0 in this case no?

Reply:
right, sorry, I confused the simple-copy and the not-simple-copy cases, I was looking at the `continue` at line 669 above